src/main/java/com/amazon/aocagent/services/EC2Service.java

```diff
@@ -66,33 +68,27 @@ public EC2Service(String region) {
   /**
    * launchInstance launches one ec2 instance.
    *
-   * @param amiID the instance amiid
+   * @param params the instance setup configuration params
    * @return InstanceID
    */
-  public Instance launchInstance(String amiID) throws Exception {
-    // tag instance for management
-    TagSpecification tagSpecification =
-        new TagSpecification()
-            .withResourceType(ResourceType.Instance)
-            .withTags(
-                new Tag(
-                    GenericConstants.EC2_INSTANCE_TAG_KEY.getVal(),
-                    GenericConstants.EC2_INSTANCE_TAG_VAL.getVal()));
-
+  public Instance launchInstance(EC2InstanceParams params) throws Exception {
     // create request
     RunInstancesRequest runInstancesRequest =
         new RunInstancesRequest()
-            .withImageId(amiID)
+            .withImageId(params.getAmiId())
             .withMonitoring(false)
             .withMaxCount(1)
             .withMinCount(1)
-            .withTagSpecifications(tagSpecification)
-            .withKeyName(GenericConstants.SSH_KEY_NAME.getVal())
+            .withTagSpecifications(params.getTagSpecification())
+            .withKeyName(params.getSshKeyName())
             .withSecurityGroupIds(
-                getOrCreateSecurityGroupByName(GenericConstants.SECURITY_GROUP_NAME.getVal()))
+                getOrCreateSecurityGroupByName(params.getSecurityGrpName()))
```

Comment:
can ec2 test pass? as it needs the "GenericConstants.SECURITY_GROUP_NAME.getVal()" which is created in setup.

Reply:
yes, I updated EC2 Testbed to build the expected EC2 launch "params".